sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresInt4Tests.java

```diff
@@ -0,0 +1,419 @@
+                        " three \n" +
+                        "-------\n" +
+                        "     3\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT 2 - -2 AS four;\n" +
+                        " four \n" +
+                        "------\n" +
+                        "    4\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT '2'::INT2 * '2'::INT2 = '16'::INT2 / '4'::INT2 AS true;\n" +
+                        " true \n" +
+                        "------\n" +
+                        " t\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT '2'::INT4 * '2'::INT2 = '16'::INT2 / '4'::INT4 AS true;\n" +
+                        " true \n" +
+                        "------\n" +
+                        " t\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT '2'::INT2 * '2'::INT4 = '16'::INT4 / '4'::INT2 AS true;\n" +
+                        " true \n" +
+                        "------\n" +
+                        " t\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT '1000'::INT4 < '999'::INT4 AS false;\n" +
+                        " false \n" +
+                        "-------\n" +
+                        " f\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 AS ten;\n" +
+                        " ten \n" +
+                        "-----\n" +
+                        "  10\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT 2 + 2 / 2 AS three;\n" +
+                        " three \n" +
+                        "-------\n" +
+                        "     3\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT (2 + 2) / 2 AS two;\n" +
+                        " two \n" +
+                        "-----\n" +
+                        "   2\n" +
+                        "(1 row)"
+        );
+    }
+
+    // Check PostgresInt2Tests::testSelectOverflow for details
+    @Test @Ignore
+    public void testSelectOverflow() {
+        String error = "INT4 out of range";
+
+        this.shouldFail("SELECT i.f1, i.f1 * '2'::INT4 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 * '2'::INT4 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 + '2'::INT2 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 + '2'::INT4 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 - '2'::INT2 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 - '2'::INT4 AS x FROM INT4_TBL i", error);
+    }
+
+    @Test @Ignore("shift operator is not supported")
```

Comment:
it won't be for a while. maybe there is a function which does the same thing?

Reply:
I took a look at the Calcite SQL reference and it doesn't seem to have a function to do the same thing. 
Removing the test.